### ISSUE TITLE
Genome inputs

### ISSUE THREAD

**Issue by skylerhar1** [JanaSperschneider/NuclearPhaser#11]
Hello Dr. Jana Sperschnieder,

I am a student researcher at Oregon State University in the laboratory of Joey Spatafora. I am working on the assembly of the dikaryotic ectomycorrhizal fungus, Rhizopogon salebrosus.  We recently read your paper “Physical separation of haplotypes in dikaryons allows benchmarking of phasing accuracy in Nanopore and HiFi assemblies with Hi‑C data”, and I would love to implement your pipeline on the data we have, but I have a couple of questions regarding the high-level genome inputs required for this pipeline 

I have multiple different assemblies, and I was wondering which should be used where in the pipeline. I have a chromosomal level genome generated from HI-C data with 11 scaffolds, a haploid fragmented genome ~ 3000 contigs; and a diploid genome with ~3300 contigs and some collapsed regions. My intuition tells me I should use my diploid genome to generate gene markers, then provide HI-C pro with the chromosomal assembly as a reference to map the HI-C fastq paired-end reads. I am very excited about the prospect of using this pipeline, so I want to make sure I use it correctly!

Thank you in advance for any assistance and please let me know if you need any additional information about the data and assemblies I have at my disposal.

Sincerely,

Skyler Har

**Comment by JanaSperschneider** [comment]
Hi Skyler,

thank you for the interest in NuclearPhaser. This software is designed to **confirm** that you have phased a dikaryotic genome and to help **find unphased regions**, so you want to give it the best approximation of a fully-phased diploid genome that you have. 

Is your chromosome-level genome with 11 scaffolds a haploid representation, or is it mostly/partially phased? 
It will be difficult to phase a highly fragmented genome with thousands of small contigs or a partially collapsed diploid genome. In any case you need to provide the same genome to the gene marker calculation and to HiC-Pro. A diploid assembly in large contigs will work best.

Best wishes,
Jana

**Comment by skylerhar1** [comment]
The chromosome-level genome with 11 scaffolds is a haploid representation. However, I phased the assembly using [Whatshapp](https://whatshap.readthedocs.io/en/latest/guide.html) and now I have 2 haploid genomes with locally phased blocks. I can combine the two genomes to make a pseudo-diploid assembly.

**Comment by JanaSperschneider** [comment]
Yes I would use the pseudo-diploid assembly. Let me know how this works :-)

**Comment by skylerhar1** [comment]
Hi Jana, 
Thank you for being so responsive. I have run through the pipeline a few times now with different inputs, but only using a mapq30 and resolution 20,000 work. For the other runs, I am not getting any results in the Plots/ directory presumably due to too many phase switches. For the original run I used a mapq of 30 and resolution of 20,000. I can't get the Rplot script to work on this data set, with the error: 
Command: 
```
p <- ggplot(data) +
  geom_area(aes(x=x, y=haplotype0), fill = "#8c510a", colour = "#8c510a", alpha=0.5) +
  stat_smooth(geom = 'area', aes(x=x, y=haplotype0), span = 1/3, alpha=0.5, fill = "#01665e") +
  xlab("Position on contig (Mb)") +
  ylab("% of Hi-C trans reads\n that link to haplotype 0")  +
  scale_x_continuous(breaks = round(seq(0, maximum, by = 500000),0)) +
  theme_bw(base_size = 48, base_family = "Helvetica") +
  theme(axis.text.x = element_text(angle=90, hjust=1)) +
  geom_segment(data = align_coords, aes(x = x, y = haplotype0, xend = y, yend = haplotype0, colour = "black"), colour = "black",  size = 3,   lineend = "round") +
  ggtitle(contig) +
  scale_x_continuous(breaks = c(seq(from = 0, to = maximum, by = 500000)),
                     labels = c(seq(from = 0, to = maximum, by = 500000))/1000000,
                     expand = c(0, 0), limits = c(0, NA))
```
Error message: 
```
Error in seq.default(0, maximum, by = 500000) : 
  'to' must be a finite number
In addition: Warning message:
In seq.default(0, maximum, by = 500000) : NAs introduced by coercion
```
 That error aside, I wanted to  discuss what my specific dataset might look like after running it through nuclearPhaser. Our Hi-C interaction map looks like this: 
![Juice_box with cov](https://user-images.githubusercontent.com/67615347/221464154-ef778a6f-ec7b-4400-8984-c9dc3107d606.png)
You'll notice a lack of interaction between chrs 5 vs 8,9,11
Our hypothesis as to why there are chromosomes with no interactions with chr5 is that they are separated by nuclear membranes within each dikaryotic cell. With no differences in chromosomal coverage or heterozygosity, we assume that each chromosome has 2 haplotypes. 

I noticed that the file: gene_bins_hic_connections.txt (attached) 
[gene_bins_hic_connections.txt](https://github.com/JanaSperschneider/NuclearPhaser/files/10835876/gene_bins_hic_connections.txt)
from the output converts interactions to a percentage. This conversion masks the absent interactions between chr5 -chr8,9,11. Will NuclearPhaser be able to detect these whole chromosomal level differences?

Thank you again for your help. If you are willing, I would love to talk about this over Zoom (or your preferred meeting method).

**Comment by JanaSperschneider** [comment]
Hi, 

sorry about the R script, which was not compatible anymore with the output file from v1.1. I fixed it so it should work now.

Two things I notice about your data:
- Your genome looks unphased with a 50:50 distribution of Hi-C links to the two haplotypes. This is a signal you would expect from a diploid, or from an dikaryotic assembly with no locally phased blocks. I don't think NuclearPhaser can help here, you would need to improve the assembly first.
- I would need to see a Hi-C contact map of your diploid assembly to see what is going on with chrs5/8,9,11. Can you plot the diploid Hi-C map you got from HiC-Pro, which you put into NuclearPhaser? This would show the normalized Hi-C contacts. If a subset of chromosomes would be separated by a nuclear membrane, wouldn't you see no contact to all other chromosome? To me this looks more like a Hi-C contact normalization issue, as there are also hardly any cis contacts on the diagonal?
- And yes, we have had a chromosome in one of our dikaryons where both copies are in the same nucleus. NuclearPhaser was able to assign both chromosomes to the correct nucleus.

Best wishes,
Jana

**Comment by skylerhar1** [comment]
Hi Jana,
Here's the diploid interaction map. The first putative haplotypes are the first 11 chromosomes, and the second haplotypes are in the position 12-22. With the current resolution, it's a little hard to see, but chr5 and its other haplotype chr16 have significantly less interactions with chr 8,9,11 and chr 19,20,22
[diploid_gap_closed_interaction_map.pdf](https://github.com/JanaSperschneider/NuclearPhaser/files/10866073/diploid_gap_closed_interaction_map.pdf)
 Here's a histogram that simplifies the results:
![chr5_interaction_histogram](https://user-images.githubusercontent.com/67615347/222299633-c9f2a11c-ccf6-477d-bb60-8aa8b44db598.jpg)

I agree that the results doesn't show strong evidence of phasing. I was thinking I could use a representative sequence of each chromosome. Each representative sequence has either has lots of phase informative SNPS, or due to an indel the sequence is only present in one haplotype. Basically, I would be trading quantity of data for quality. Do you think this strategy could work?

Our current hypothesis of the chromosomal distribution is this, where each parenthesis representing one nucleus :
(1,2,3,4,6,7,8,8,9,9,10,11,11) X (1,2,3,4,5,5,6,7,10)
1,2,3,4,6,7,10 are universal
5,8,9,11 are nucleus specific.

Many Thanks,
Skyler
